### PR TITLE
Add test case for ShowSQLParserRuleExecutor

### LINKED ISSUE #14806: Add test case for ShowSQLParserRuleExecutor
### Background
ShowAuthorityRuleExecutor is the executor for DistSQL `SHOW SQL_PARSER RULE`.
Document: https://shardingsphere.apache.org/document/current/en/user-manual/shardingsphere-proxy/distsql/syntax/ral/

### How to
Refer to the test case for `ShowTransactionRuleExecutor` : [ShowTransactionRuleExecutorTest](https://github.com/apache/shardingsphere/blob/master/shardingsphere-proxy/shardingsphere-proxy-backend/src/test/java/org/apache/shardingsphere/proxy/backend/text/distsql/ral/common/show/ShowTransactionRuleExecutorTest.java).

### PR DESCRIPTION
**Pull request by Liangda-w**
Fixes #14806 

Changes proposed in this pull request:
- Add test case for ShowSQLParserRuleExecutor

### CODE REVIEW

File: shardingsphere-proxy/shardingsphere-proxy-backend/src/test/java/org/apache/shardingsphere/proxy/backend/text/distsql/ral/common/show/ShowSQLParserRuleExecutorTest.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+import java.util.List;
+
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Mockito.RETURNS_DEEP_STUBS;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
```
Comment (terrymanu) [inline shardingsphere-proxy/shardingsphere-proxy-backend/src/test/java/org/apache/shardingsphere/proxy/backend/text/distsql/ral/common/show/ShowSQLParserRuleExecutorTest.java:43]: It is unnecessary If do not use `@Mock` to create mocked objects, could you remove it?

File: shardingsphere-proxy/shardingsphere-proxy-backend/src/test/java/org/apache/shardingsphere/proxy/backend/text/distsql/ral/common/show/ShowSQLParserRuleExecutorTest.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+public final class ShowSQLParserRuleExecutorTest {
+
+    private final ShowSQLParserRuleExecutor executor = new ShowSQLParserRuleExecutor();
```
Comment (terrymanu) [inline shardingsphere-proxy/shardingsphere-proxy-backend/src/test/java/org/apache/shardingsphere/proxy/backend/text/distsql/ral/common/show/ShowSQLParserRuleExecutorTest.java:46]: Is it necessary to declare as global filed, How about change it to local variable?

**Review by terrymanu — CHANGES_REQUESTED** [review]

File: shardingsphere-proxy/shardingsphere-proxy-backend/src/test/java/org/apache/shardingsphere/proxy/backend/text/distsql/ral/common/show/ShowSQLParserRuleExecutorTest.java (hunk 1 above)
Comment (Liangda-w) [reply shardingsphere-proxy/shardingsphere-proxy-backend/src/test/java/org/apache/shardingsphere/proxy/backend/text/distsql/ral/common/show/ShowSQLParserRuleExecutorTest.java:43]: thanks for the note, I'll remove it!

File: shardingsphere-proxy/shardingsphere-proxy-backend/src/test/java/org/apache/shardingsphere/proxy/backend/text/distsql/ral/common/show/ShowSQLParserRuleExecutorTest.java (hunk 2 above)
Comment (Liangda-w) [reply shardingsphere-proxy/shardingsphere-proxy-backend/src/test/java/org/apache/shardingsphere/proxy/backend/text/distsql/ral/common/show/ShowSQLParserRuleExecutorTest.java:46]: yes, that's better for sure

**Comment by RaigorJiang** [comment]
@Liangda-w  thank you for your contribution,
and @terrymanu Thanks for your review.

**Review by RaigorJiang — APPROVED** [review]

**Review by terrymanu — APPROVED** [review]

**Pull request merged**